new metrics

dbernstein1 opened the pull request:
[PMM-XXXX](https://jira.percona.com/browse/PMM-XXXX) (optional, if ticket reported)

- [ ] Links to other linked pull requests (optional).

---

- [ ] Tests passed.
- [ ] Fix conflicts with target branch.
- [ ] Update jira ticket description if needed.
- [ ] Attach screenshots/console output to confirm new behavior to jira ticket, if applicable.

Once all checks pass and the code is ready for review, please add `pmm-review-exporters` team as the reviewer. That would assign people from the review team automatically. Report any issues on our [Forum](https://forums.percona.com) or [Discord](https://per.co.na/discord).

it-percona-cla commented:
[![CLA assistant check](https://cla-assistant.percona.com:443/pull/badge/not_signed)](https://cla-assistant.percona.com:443/percona/mongodb_exporter?pullRequest=576) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.percona.com:443/percona/mongodb_exporter?pullRequest=576) before we can accept your contribution.<br/><hr/>**Daniel Bernstein** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.percona.com:443/check/percona/mongodb_exporter?pullRequest=576) it.</sub>

BupycHuk commented on exporter/v1_compatibility.go:
```diff
@@ -1200,9 +1222,118 @@ func chunksTotal(ctx context.Context, client *mongo.Client) (prometheus.Metric,
+func chunksTotal(ctx context.Context, client *mongo.Client) (prometheus.Metric, error) {
+	n, err := client.Database("config").Collection("chunks").CountDocuments(ctx, bson.M{})
+	if err != nil {
+@ -1200,9 +1317,26 @@ func chunksTotal(ctx context.Context, client *mongo.Client) (prometheus.Metric,
```
this line looks pretty weird

dbernstein1 replied:
I tried running make format but got these issues:

go mod tidy
go mod tidy: go.mod file indicates go 1.17, but maximum supported version is 1.16
make: *** [format] Error 1

BupycHuk replied:
What version of golang do you have locally?

dbernstein1 replied:
1.16.5

BupycHuk replied:
Please update to at least 1.17.

BupycHuk commented:
Hi, @dbernstein1 thanks for the contribution. Could you provide information on what this PR is about and what problem you are fixing here? Thanks

dbernstein1 commented:
@BupycHuk 

1. added the following metrics: 
- collections with chunks currently being balanced among shards (mongodb_mongos_sharding_migrating)
- System balancer running on/off indictation (mongodb_mongos_sharding_balancer_running)
- chunk size in megabytes (mongodb_mongos_sharding_chunk_size_mb)

2. Included a label with the db/collection in mongodb_mongos_sharding_shard_chunks_total to see how chunks are balanced across shards in a given collection

3. Fixed a bug in balancerEnabled mongodb_mongos_sharding_balancer_enabled always returned 1 even when balanced was disabled

BupycHuk commented:
@dbernstein1 please sign the CLA

dbernstein1 commented:
@BupycHuk I already did but it is showing up that I didn’t 
<img width="1419" alt="Screen Shot 2022-10-21 at 3 53 01 PM" src="https://user-images.githubusercontent.com/44444136/197301537-8fe20cd5-ae27-4a4a-86c3-d25cddeb7e1c.png">

dbernstein1 commented:
<img width="1499" alt="Screen Shot 2022-10-21 at 3 55 25 PM" src="https://user-images.githubusercontent.com/44444136/197301974-71cade4e-e00a-42c3-acbc-afbb439f02ba.png">

artemgavrilov commented:
Hi @dbernstein1 your commits signed with `dabernst@cisco.com` email address, but your GH account has different address. Because of that CLA check is not passing.

BupycHuk commented:
@dbernstein1 please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user)

dbernstein1 commented:
@BupycHuk I added dabernst@cisco.com to my github account. Tried signing the CLA again

<img width="953" alt="Screenshot 2022-12-05 at 6 36 06 PM" src="https://user-images.githubusercontent.com/44444136/205794889-1251e7ac-f304-49a5-a63d-268731d523e4.png">

artemgavrilov commented:
@dbernstein1 @BupycHuk CLA problem has gone.

BupycHuk reviewed (changes requested):
Hi @dbernstein1, Thank you for your contribution and fixing a problem with CLA.

v1_compatibilty collector is supposed to be a collector to keep compatibility with mongodb_exporter v1. For new metrics it's better to create a new collector.

BupycHuk commented:
closed as outdated
